File: API/user-service/src/main/java/com/api/userservice/service/TaskService.java
package com.api.userservice.service;

import java.util.List;

import com.api.userservice.DTO.TaskDTO;
import com.api.userservice.model.Task;
import com.api.userservice.model.User;

public interface TaskService {
    List<TaskDTO> getTasksByUserId(Long userId);

    List<TaskDTO> getTasksByUserId(Long userId, boolean completed);

    boolean setCompletedTask(Long userId, Long taskId, boolean completed);

    void addNewTask(User user, String header, String description, String deadline);

    Task getTaskById(Long taskId);
}

File: client/app/src/main/java/com/src/onboarding/data/remote/service/TaskService.kt
package com.src.onboarding.data.remote.service

import com.src.onboarding.data.remote.model.task.TaskResponse
import com.src.onboarding.di.NetworkModule
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query

interface TaskService {
    @GET("${NetworkModule.USER_SERVICE_BASE_URL}/task/get-tasks-by-user-id")
    suspend fun getTasksByUserId(@Query("userId") userId: Long): Response<List<TaskResponse>>

    @GET("${NetworkModule.USER_SERVICE_BASE_URL}/task/set-competed-task")
    suspend fun setCompletedTask(
        @Query("taskId") taskId: Long,
        @Query("completed") completed: Boolean
    ): Response<Unit>

    @GET("${NetworkModule.USER_SERVICE_BASE_URL}/task/add-new-task")
    suspend fun addTask(
        @Query("userId") userId: Long,
        @Query("header") header: String,
        @Query("description") description: String,
        @Query("deadline") deadline: String
    ): Response<Unit>

    @GET("${NetworkModule.USER_SERVICE_BASE_URL}/task/get-task-filter")
    suspend fun getTasksByIsCompleted(@Query("completed") completed: Boolean): Response<List<TaskResponse>>
}